src/execution_plan/execution_plan.h

```diff
@@ -21,8 +21,6 @@ struct ExecutionPlan {
 	QueryGraph *query_graph;            // QueryGraph representing all graph entities in this segment.
 	FT_FilterNode *filter_tree;         // FilterTree containing filters to be applied to this segment.
 	QueryGraph **connected_components;  // Array of all connected components in this segment.
-	// NOTE - segments and segment_count are only stored for proper freeing.
-	int segment_count;                  // Number of ExecutionPlan segments.
```

Comment:
`segments` was a simple heap-allocated array. I changed to `arr.h` array so we can get the length from it.
I changed it to have a uniform approach to arrays in the execution plan struct since `connected_components` was `arr.h` array. It seems better and cleaner to code this way.